lib/main_publiccloud.pm

```diff
@@ -77,47 +77,68 @@ sub load_publiccloud_consoletests {
+    elsif (get_var('PUBLIC_CLOUD_ACCNET')) {
+        loadtest 'publiccloud/az_accelerated_net';
+    }
+    elsif (get_var('PUBLIC_CLOUD_CHECK_BOOT_TIME')) {
+        loadtest "publiccloud/boottime";
+    }
+    elsif (get_var('PUBLIC_CLOUD_FIO')) {
+        loadtest 'publiccloud/storage_perf';
+    }
+    elsif (get_var('PUBLIC_CLOUD_CONSOLE_TESTS')) {
+        my $args = OpenQA::Test::RunArgs->new();
+        loadtest "publiccloud/prepare_instance", run_args => $args;
+        loadtest "publiccloud/register_system", run_args => $args;
+        loadtest "publiccloud/ssh_interactive_start", run_args => $args;
+        load_extra_tests_prepare();
+        load_publiccloud_consoletests();
+        loadtest("publiccloud/ssh_interactive_end", run_args => $args);
+    }
+    elsif (get_var('PUBLIC_CLOUD_IMAGE_LOCATION')) {
+        loadtest "publiccloud/upload_image";
+    }
+}
+
+sub load_publiccloud_tools {
+    loadtest 'autoyast/prepare_profile';
+    loadtest 'installation/bootloader';
+    loadtest 'autoyast/installation';
+    loadtest 'publiccloud/prepare_tools';
+    loadtest 'shutdown/shutdown';
+}
+
+sub load_publiccloud_download_repos {
+    loadtest 'publiccloud/download_repos';
+    loadtest 'shutdown/shutdown';
+}
+
 sub load_publiccloud_tests {
-    loadtest 'boot/boot_to_desktop';
-    if (check_var('PUBLIC_CLOUD_DOWNLOAD_TESTREPO', 1)) {
-        loadtest 'publiccloud/download_repos';
-        loadtest 'shutdown/shutdown';
+    if (check_var('PUBLIC_CLOUD_TOOLS_REPO', 1)) {
+        load_publiccloud_tools();
     }
-    elsif (get_var('PUBLIC_CLOUD_QAM')) {
-        load_maintenance_publiccloud_tests();
-    } else {
-        if (get_var('PUBLIC_CLOUD_PREPARE_TOOLS')) {
-            loadtest "publiccloud/prepare_tools";
-        }
-        elsif (get_var('PUBLIC_CLOUD_IMG_PROOF_TESTS')) {
-            loadtest "publiccloud/img_proof";
-        }
-        elsif (get_var('PUBLIC_CLOUD_LTP')) {
-            loadtest 'publiccloud/run_ltp';
-        }
-        elsif (get_var('PUBLIC_CLOUD_SLES4SAP')) {
-            loadtest 'publiccloud/sles4sap';
-        }
-        elsif (get_var('PUBLIC_CLOUD_ACCNET')) {
-            loadtest 'publiccloud/az_accelerated_net';
-        }
-        elsif (get_var('PUBLIC_CLOUD_CHECK_BOOT_TIME')) {
-            loadtest "publiccloud/boottime";
-        }
-        elsif (get_var('PUBLIC_CLOUD_FIO')) {
-            loadtest 'publiccloud/storage_perf';
-        }
-        elsif (get_var('PUBLIC_CLOUD_CONSOLE_TESTS')) {
-            my $args = OpenQA::Test::RunArgs->new();
-            loadtest "publiccloud/prepare_instance", run_args => $args;
-            loadtest "publiccloud/register_system", run_args => $args;
-            loadtest "publiccloud/ssh_interactive_start", run_args => $args;
-            load_extra_tests_prepare();
-            load_publiccloud_consoletests();
-            loadtest("publiccloud/ssh_interactive_end", run_args => $args);
+    else {
+        loadtest 'boot/boot_to_desktop';
```

Comment:
no because `load_create_publiccloud_tools_image` has `loadtest 'installation/bootloader'`

Reply:
https://openqa.suse.de/group_overview/276 may give you an idea

Reply:
Ahhhhh, got it, makes sense.

Reply:
Please add a comment to the if branch, that makes this clear. Otherwise we will have the same discussion in a week from now.

Something like

    if (check_var('PUBLIC_CLOUD_TOOLS_REPO', 1)) {
        # 'installation/bootloader' is loaded instead of 'boot_to_desktop' within this subroutine.

Or something similar

Reply:
The comment wasnt make sense IMO, because it is obvious why the `load_create_publiccloud_tools_image` is not using `boot_to_desktop` and why then, it is defined in the *else*. However i thought that it would be a good idea to provide a small description. So *This is ready for review* from my side